src/bscPlugin/validation/ScopeValidator.ts

```diff
@@ -351,4 +360,108 @@ export class ScopeValidator {
         });
         event.scope.addDiagnostics(diagnostics);
     }
+
+    protected detectInvalidFunctionCalls(event: OnScopeValidateEvent) {
+        const diagnostics: BsDiagnostic[] = [];
+        //get a list of all callables, indexed by their lower case names
+        let callableContainerMap = event.scope.getCallableContainerMap();
+
+        event.scope.enumerateBrsFiles((file) => {
+            diagnostics.push(...this.diagnosticDetectInvalidFunctionCalls(event.scope, file, callableContainerMap));
+        });
+        event.scope.addDiagnostics(diagnostics);
+    }
+
+
+    /**
+    * Find functions with either the wrong type of parameters, or the wrong number of parameters
+    */
+    private diagnosticDetectInvalidFunctionCalls(scope: Scope, file: BscFile, callableContainersByLowerName: CallableContainerMap) {
```

Comment:
Rename this function. The word `diagnostic` is a holdover from the scope file. Maybe `detectInvalidFunctionCallsForFile`?